tracecat/tables/csv_table.py

```diff
@@ -0,0 +1,96 @@
+from typing import Any
+
+import polars as pl
+from pydantic import BaseModel
+
+from tracecat.tables.enums import SqlType
+
+
+class InferredColumn(BaseModel):
+    """Information about an inferred column."""
+
+    name: str
+    type: SqlType
+    sample_value: Any = None
+
+
+class SchemaInferenceService:
```

Comment:
wouldn't name this a service - currently services abstract away some kind of external IO (db, temporal)